Fix ONNX frontend Reshape behavior

trulyspinach opened the pull request:
When performing a 1D to 4D reshape, the current implementation directly perform an in order remap which leads to an unexpected transpose from NCHW(1D) to NHWC thus make the accelerator outputs different result from other ONNX inference engine.

This can be easily reproduced with the following ONNX model:
[reshaper.onnx.zip](https://github.com/tensil-ai/tensil/files/8547055/reshaper.onnx.zip)


<img width="251" alt="Screen Shot 2022-04-23 at 9 48 29 PM" src="https://user-images.githubusercontent.com/8978490/164905746-0eaf5279-14f6-4b80-9aef-7811571f2ffa.png">

Generated with the following PyTorch code:
```
class Reshaper(nn.Module):
    def __init__(self):
        super(Reshaper, self).__init__()

        self.conv1 = nn.Conv2d(2, 2, 3, stride=1, padding=0)
        self.up = nn.Upsample(scale_factor=2, mode='bilinear', align_corners=False)

    def forward(self, z):
        out = z
        out = z.reshape(z.shape[0], 2,2,2)
        # out = self.up(out)
        return out
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/tensil-ai/tensil?pullRequest=45) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/tensil-ai/tensil?pullRequest=45) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/tensil-ai/tensil?pullRequest=45) it.</sub>

petrohi reviewed (changes requested):
Looks great! Two suggestions:

- Please format the code according to project's `.scalafmt.conf`.
- (Optional) Please consider adding compiler test that captures this behavior. You'd need to add your test model to https://github.com/tensil-ai/tensil-models. For an example take a look at https://github.com/tensil-ai/tensil/blob/main/tools/test/src/tools/CompilerSpec.scala#L1254.

trulyspinach commented:
@petrohi Thanks!
I've updated the format and added a test case that checks for the output. 
Could you push the model file here into [tensil-models](https://github.com/tensil-ai/tensil-models)?
[reshape_1d_4d.onnx.zip](https://github.com/tensil-ai/tensil/files/8562381/reshape_1d_4d.onnx.zip)

petrohi commented on tools/test/src/tools/Reshape.scala:
```diff
@@ -0,0 +1,62 @@
+    assert(rmse.compute < dataType.error)
+  }
+
+  private val Golden = Seq(
```
I suggest asserting on the entire result including zeroes.

trulyspinach replied:
Good point! I've fixed it now it checks the full output.

petrohi reviewed (changes requested):
This is awesome! Please see my suggestion.

petrohi commented:
I added test model to `tensil-models`.
